chain/chain/src/store.rs

```diff
@@ -2393,32 +2399,25 @@ impl<'a> ChainStoreUpdate<'a> {
                     entry.insert(hash_set);
                 }
             };
+
             // Increase transaction refcounts for all included txs
             for tx in chunk.transactions.iter() {
-                let mut refs = self.chain_store.get_tx_refcount(&tx.get_hash())?;
-                refs += 1;
-                store_update.set_ser(ColTransactionRefCount, tx.get_hash().as_ref(), &refs)?;
+                let mut refcount = self.chain_store.get_tx_refcount(&tx.get_hash())?;
+                refcount += 1;
+                store_update.set_ser(ColTransactionRefCount, tx.get_hash().as_ref(), &refcount)?;
```

Comment:
Do we ensure that this does not lead us to a race condition?

Reply:
We cannot increase and decrease Tx Refcount in one store update operation. While increasing, we cannot add same Tx twice. There is also a comment in `finalize()` method explaining this.